velox/functions/prestosql/BinaryFunctions.h

```diff
@@ -436,4 +436,107 @@ struct ToIEEE754Bits32 {
   }
 };
 
+template <bool lpad, bool isAscii, typename TOut, typename TIn>
+FOLLY_ALWAYS_INLINE void
+pad(TOut& output, const TIn& input, const int64_t size, const TIn& padding) {
+  static constexpr size_t padMaxSize = 1024 * 1024; // 1MB
+
+  VELOX_USER_CHECK(
+      size >= 0 && size <= padMaxSize,
+      "pad size must be in the range [0..{})",
+      padMaxSize);
+  VELOX_USER_CHECK(padding.size() > 0, "padding bytes must not be empty");
+
+  int64_t inputSize = input.size();
+  // Input has at most size length, truncate it if necessary
+  // and return it as the result.
+  if (UNLIKELY(inputSize >= size)) {
+    output.resize(size);
+    std::memcpy(output.data(), input.data(), size);
```

Comment:
Can we avoid this memcpy anyhow ? If TIn is StringView then we can just slice over the original I would think.